src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -218,9 +218,9 @@ const TopicFragment = ({
     renderDiscoveryItem({
       from: isFirstTimeOpen
         ? FROM_FOLLOWED_TOPIC_INITIAL
-        : index > newMapFollowedTopics.length
-        ? FROM_UNFOLLOWED_TOPIC
-        : FROM_FOLLOWED_TOPIC,
+        : item.following || item.user_id_follower !== null
+        ? FROM_FOLLOWED_TOPIC
+        : FROM_UNFOLLOWED_TOPIC,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new logic for determining the `from` value seems to be more accurate as it checks whether the item is followed or if it has a follower user ID. However, it's important to ensure that `item.following` and `item.user_id_follower` are always defined to avoid potential issues with undefined values. If these properties could potentially be undefined, consider adding additional checks.

```diff
- from: isFirstTimeOpen
-   ? FROM_FOLLOWED_TOPIC_INITIAL
-   : item.following || item.user_id_follower !== null
-   ? FROM_FOLLOWED_TOPIC
-   : FROM_UNFOLLOWED_TOPIC,
+ from: isFirstTimeOpen
+   ? FROM_FOLLOWED_TOPIC_INITIAL
+   : (item.following !== undefined && item.following) || (item.user_id_follower !== undefined && item.user_id_follower !== null)
+   ? FROM_FOLLOWED_TOPIC
+   : FROM_UNFOLLOWED_TOPIC,
```
This change ensures that we're not just checking for truthiness, but also that the properties are not undefined before making the comparison.


<!-- This is an auto-generated comment by OSS CodeRabbit -->